x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -196,30 +205,32 @@ export class AlertsClient {
     }
   }
 
-  public async getAuthorizedAlertsIndices(featureIds: string[]): Promise<string[] | undefined> {
+  public async getAuthorizedAlertsIndices(
+    featureIds: readonly ValidFeatureIds[]
+  ): Promise<string[] | undefined> {
     const augmentedRuleTypes = await this.authorization.getAugmentRuleTypesWithAuthorization(
-      featureIds,
+      featureIds as string[],
       [ReadOperations.Find, ReadOperations.Get, WriteOperations.Update],
       AlertingAuthorizationEntity.Alert
     );
 
     // As long as the user can read a minimum of one type of rule type produced by the provided feature,
     // the user should be provided that features' alerts index.
     // Limiting which alerts that user can read on that index will be done via the findAuthorizationFilter
-    const authorizedFeatures = new Set();
+    const authorizedFeatures = new Set<string>();
     for (const ruleType of augmentedRuleTypes.authorizedRuleTypes) {
       authorizedFeatures.add(ruleType.producer);
     }
 
+    const isValidFeatureId = (a: string): a is ValidFeatureIds =>
+      // @ts-expect-error Argument of type 'string' is not assignable to parameter of type '"apm" | "observability" | "siem"'
+      validFeatureIds.includes(a);
```

Comment:
If you make the map changes I show above, this can change to:
```ts
const isValidFeatureId = (a: string): a is ValidFeatureId => Object.keys(mapConsumerToIndexName).includes(a);
```
so that the "@ts-expect-error" can be removed.